pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -348,10 +354,22 @@ func checkCertValidity(certBytes, keyBytes []byte, ipAddresses []string, nodeUID
 		}
 	}
 
-	// TODO(marun) Check that the certificate was issued by the CA
+	if ok := lessThanMinimumDuration(leafCert, defaultMinDurationPercent); ok {
+		return fmt.Sprintf("less than %d%% duration remaining", int64(defaultMinDurationPercent*100)), nil
+	}
 
-	// TODO(marun) Check that the certificate is not expired or due for replacement
+	// TODO(marun) Check that the certificate was issued by the CA
 
 	// Cert is valid
 	return "", nil
 }
+
+// lessThanMinimumDuration indicates whether the provided cert has less
+// than the provided minimum percentage of its duration remaining.
+func lessThanMinimumDuration(cert *x509.Certificate, minDurationPercent float64) bool {
```

Comment:
One suggestion: if you pass directly the expiry and duration, this could be nicely unit tested with various test cases.

for example:

```go
func lessThanMinimumDuration(expiry, duration time.Time, minDurationPercent float64)
```

and call above with:
```go
lessThanMinimumDuration(leafCert.NotAfter, leafCert.NotBefore, defaultMinDurationPercent)
```

Reply:
Done [#lawofdemeter](https://en.wikipedia.org/wiki/Law_of_Demeter)